src/EditorFeatures/Core.Wpf/InlineDiagnostics/InlineDiagnosticsFormatDefinition.cs

```diff
@@ -0,0 +1,84 @@
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.InlineDiagnostics
+{
+    internal sealed class ClassificationTypeDefinitions
+    {
+        [Export]
+        [Name(InlineDiagnosticsTag.TagID + PredefinedErrorTypeNames.SyntaxError)]
+        [BaseDefinition(PredefinedClassificationTypeNames.FormalLanguage)]
+        internal ClassificationTypeDefinition InlineErrorsErrorTypeDefinition;
+
+        [Export(typeof(EditorFormatDefinition))]
+        [Name(InlineDiagnosticsTag.TagID + PredefinedErrorTypeNames.SyntaxError)]
+        [Order(After = LanguagePriority.NaturalLanguage, Before = LanguagePriority.FormalLanguage)]
+        [UserVisible(true)]
+        private class InlineDiagnosticsErrorFormatDefinition : EditorFormatDefinition
+        {
+            [ImportingConstructor]
+            [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+            public InlineDiagnosticsErrorFormatDefinition()
+            {
+                DisplayName = EditorFeaturesResources.Inline_Diagnostics_Error;
+                BackgroundBrush = new SolidColorBrush(Color.FromRgb(252, 62, 54));
+                ForegroundBrush = new SolidColorBrush(Color.FromRgb(0, 0, 0));
+            }
+        }
+
+        [Export]
+        [Name(InlineDiagnosticsTag.TagID + PredefinedErrorTypeNames.Warning)]
+        [BaseDefinition(PredefinedClassificationTypeNames.FormalLanguage)]
+        internal ClassificationTypeDefinition InlineErrorsWarningTypeDefinition;
+
+        [Export(typeof(EditorFormatDefinition))]
+        [Name(InlineDiagnosticsTag.TagID + PredefinedErrorTypeNames.Warning)]
+        [Order(After = LanguagePriority.NaturalLanguage, Before = LanguagePriority.FormalLanguage)]
+        [UserVisible(true)]
+        private class InlineDiagnosticsWarningFormatDefinition : EditorFormatDefinition
+        {
+            [ImportingConstructor]
+            [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+            public InlineDiagnosticsWarningFormatDefinition()
+            {
+                DisplayName = EditorFeaturesResources.Inline_Diagnostics_Warning;
+                BackgroundBrush = new SolidColorBrush(Color.FromRgb(149, 219, 125));
+                ForegroundBrush = new SolidColorBrush(Color.FromRgb(0, 0, 0));
+            }
+        }
+
+        [Export]
+        [Name(InlineDiagnosticsTag.TagID + EditAndContinueErrorTypeDefinition.Name)]
+        [BaseDefinition(PredefinedClassificationTypeNames.FormalLanguage)]
+        internal ClassificationTypeDefinition InlineErrorsRudeEditTypeDefinition;
+
+        [Export(typeof(EditorFormatDefinition))]
+        [Name(InlineDiagnosticsTag.TagID + EditAndContinueErrorTypeDefinition.Name)]
+        [Order(After = LanguagePriority.NaturalLanguage, Before = LanguagePriority.FormalLanguage)]
+        [UserVisible(true)]
+        private class InlineDiagnosticsRudeEditFormatDefinition : EditorFormatDefinition
```

Comment:
do you have a picture of what this looks like?

Reply:
![image](https://user-images.githubusercontent.com/40616383/125838879-f17c305f-6b68-43a1-8365-35dd744374a3.png)